homework/index.js

```diff
@@ -29,23 +31,127 @@
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, {
+      text: title,
+    });
+    createAndAppend('td', tr, { text: repoText });
+    return tr;
+  }
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const date = new Date(repo.updated_at);
+    const li = createAndAppend('li', ul, { class: 'li-root' });
+    const table = createAndAppend('table', li);
+    const firstRow = addRow(table, 'Repository:', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+
+    addRow(table, 'Description:', repo.description, 'td');
+    addRow(table, 'Forks:', repo.forks, 'td');
+    addRow(table, 'Updated:', date.toLocaleDateString(), 'td');
+  }
+
+  function createOptions(repo, parent, index) {
+    createAndAppend('option', parent, {
+      text: repo.name,
+      value: index,
+    });
+  }
+  function selections(selection, data, parent) {
+    const ul = createAndAppend('ul', parent, {
+      class: 'ul-root',
+    });
+    const repo = data[selection];
+    renderRepoDetails(repo, ul);
+  }
+  function createContributorsSection(url, parent) {
+    fetchJSON(url).then(data => {
+      data.forEach(repo => {
+        const li = createAndAppend('li', parent);
+        createAndAppend('img', li, {
+          src: repo.avatar_url,
+          class: 'avatar',
+        });
+        createAndAppend('a', li, {
+          text: repo.login,
+          href: repo.html_url,
+          target: '_blank',
+          class: 'cont-name',
+        });
+        createAndAppend('p', li, {
+          text: repo.contributions,
+        });
+      });
+    });
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'hyf-title',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+      class: 'main-title',
+    });
+    const mainSection = createAndAppend('main', root, {
+      class: 'main-container',
+    });
+    const repoContainerSection = createAndAppend('section', mainSection, {
+      class: 'repo-container',
+    });
+    const ul = createAndAppend('ul', repoContainerSection, {
+      class: 'ul-root',
+    });
+    const contributorsContainerSection = createAndAppend(
+      'section',
+      mainSection,
+      {
+        class: 'contributor-container',
+      },
+    );
+    const p = createAndAppend('p', contributorsContainerSection, {
+      class: 'contributors-title',
+    });
+    p.innerHTML = 'Contributions';
+
+    const ulCont = createAndAppend('ul', contributorsContainerSection, {
+      class: 'ul-cont-root',
+    });
+
+    const select = createAndAppend('select', header, {
+      class: 'selection-module',
+    });
+
+    fetchJSON(url)
+      .then(data => {
```

Comment:
Here again, take away the guessing by using a descriptive name for the data you receive from `fetchJSON()`. In this case, you are fetching an array of repository objects, therefore the name `repositories` or `repos` would seem appropriate.